lib/web_ui/lib/src/engine/history/js_url_strategy.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.10
+part of engine;
+
+typedef _PathGetter = String Function();
+
+typedef _StateGetter = dynamic Function();
+
+typedef _OnPopState = ui.VoidCallback Function(html.EventListener);
+
+typedef _StringToString = String Function(String);
+
+typedef _StateOperation = void Function(
+    dynamic state, String title, String url);
+
+typedef _HistoryMove = Future<void> Function(int count);
+
+/// Given a Dart implementation of URL strategy, it converts it to a JavaScript
+/// URL strategy that be passed through JS interop.
+JsUrlStrategy? convertToJsUrlStrategy(UrlStrategy? strategy) {
```

Comment:
This is what I tried first. But then decided against it because it requires app developers to do JS classes, etc. I want app developers to stay 100% in Dart and not even realize that there's JS interop going on under the hood.

Is there a way to do that with your suggestion? I'm not able to see it but I could be wrong.

Reply:
This function can be deleted now, right?